Remove strict_apply option, make the service account name configurable

ribbybibby opened the pull request.

ffilippopoulos commented:
lgtm! Do you think we can add the rbac and SA manifests for the other namespaces as well under manifests?

george-angel commented on README.md:
```diff
@@ -10,6 +10,8 @@ kube-applier runs as a Pod in your cluster and watches the [Git repo](#mounting-
 
 Whenever a new commit to the repo occurs, or at a [specified interval](#run-interval), kube-applier performs a "run", issuing [kubectl apply](https://kubernetes.io/docs/user-guide/kubectl/v1.6/#apply) commands with pruning at namespace level. The convention is that level 1 subdirs of REPO_PATH represent k8s namespaces: the name of the dir is the same as the namespace and the dir contains manifest ...
 
+kube-applier will try to fetch a Service Account under each namespace (as defined by the environment variable `SERVICE_ACCOUNT_NAME`) and will use it to run kubectl commands. It will error for all namespaces that do not contain a matching SA.
```
`ACTING_SERVICE_ACCOUNT_NAME`?

ribbybibby replied:
Hm, I suppose this disambiguates it from the SA of the actual kube-applier service. Although I think the distinction is clear enough without it.

I think I prefer the sound of `ASSUMED_SERVICE_ACCOUNT_NAME`, but I don't know if this is an accurate description of what is actually happening.

george-angel replied:
Happy for it to be something else, but if you have an application, and it has a flag called "service account name" - my first thought would be - the service account the application runs with, not the SA it gets from other namespaces to apply them.

ribbybibby replied:
Changed.

ribbybibby commented:
@ffilippopoulos if I understand kustomize, you mean adding a separate base for the other namespaces that does the RBAC and SA bits?

ffilippopoulos commented:
@ribbybibby precisely, that was my idea. I don't know if it is good enough to be included here, just pitching it..

george-angel commented:
I think its a good idea, then instruction can be to import the "main application" base where you want KA to run, and "managed namespaces" base for the SA/rbac.

ribbybibby commented:
@george-angel @ffilippopoulos I've added a base for the managed namespaces. I moved the existing base under `manifests/main` and created a new folder: `manifests/managed-namespace`.

george-angel commented on manifests/auth/example/auth-kube-applier-patch.yaml:
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+metadata:
+  name: kube-applier-assumed
+  namespace: managed-namespace-name
```
I would remove these

hectorhuertas commented on manifests/auth/example/auth-kube-applier-patch.yaml:
```diff
@@ -0,0 +1,22 @@
+---
```
I'd remove this

hectorhuertas commented on manifests/auth/base/auth-kube-applier.yaml:
```diff
@@ -0,0 +1,24 @@
+---
```
I'd remove this

hectorhuertas commented on manifests/auth/base/auth-kube-applier.yaml:
```diff
@@ -0,0 +1,24 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: kube-applier-sa-get
```
This could be called just `kube-applier`?

ribbybibby replied:
I quite like the fact that it isn't just `kube-applier` - there's too much 'kube-applier' over all the objects here, ha. 

What about `kube-applier-role`? Leaves it open to be appended to or amended.

george-angel replied:
Its already a "role" - its like having a variable called "error_count_variable"

hectorhuertas replied:
There is a lot of `kube-applier` around, but that's common in kube ecosystem. If there aren't any other KA roles, I'd lend over it being called just `kube-applier`

ribbybibby replied:
`kube-applier` it is
